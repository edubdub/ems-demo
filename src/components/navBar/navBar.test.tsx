import * as React from 'react'
import { shallow } from 'enzyme'
import NavBar from './navBar'

it('renders without crashing', () => {
  expect(
    shallow(
      <NavBar
        date={new Date(0)}
        fuzzySearch={true}
        height={64}
        heightRealized={() => null}
        onDateChange={() => null}
        onFuzzySearchChange={() => null}
        onPlusClick={() => null}
        onSearchTermChange={() => null}
        searchTerm="test"
      />
    )
  ).toMatchSnapshot()
})
