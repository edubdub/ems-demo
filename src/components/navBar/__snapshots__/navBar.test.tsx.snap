// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    date={2017-06-13T11:41:20.000Z}
    fuzzySearch={true}
    height={64}
    heightRealized={[Function]}
    onDateChange={[Function]}
    onFuzzySearchChange={[Function]}
    onPlusClick={[Function]}
    onSearchTermChange={[Function]}
    searchTerm="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "position": "fixed",
                        "width": "100%",
                      }
        }
>
        <AppBar
                iconElementLeft={
                        <div
                                style={
                                        Object {
                                                "height": "100%",
                                                "maxWidth": 50,
                                              }
                                }
                        >
                                <DatePicker
                                        autoOk={false}
                                        container="inline"
                                        disableYearSelection={false}
                                        disabled={false}
                                        firstDayOfWeek={1}
                                        hideCalendarDate={false}
                                        id="calendarNav"
                                        inputStyle={
                                                Object {
                                                        "color": "white",
                                                      }
                                        }
                                        mode="landscape"
                                        onChange={[Function]}
                                        openToYearSelection={false}
                                        style={Object {}}
                                        value={2017-06-13T11:41:20.000Z}
                                />
                        </div>
                }
                iconElementRight={
                        <span>
                                <FlatButton
                                        disabled={false}
                                        fullWidth={false}
                                        label={
                                                <FontIcon
                                                        className="fa fa-search"
                                                        color="#ffffff"
                                                        onMouseEnter={[Function]}
                                                        onMouseLeave={[Function]}
                                                />
                                        }
                                        labelPosition="after"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                Object {
                                                        "height": "100%",
                                                      }
                                        }
                                />
                                <FlatButton
                                        disabled={false}
                                        fullWidth={false}
                                        label={
                                                <FontIcon
                                                        className="fa fa-plus"
                                                        color="#ffffff"
                                                        onMouseEnter={[Function]}
                                                        onMouseLeave={[Function]}
                                                />
                                        }
                                        labelPosition="after"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                Object {
                                                        "height": "100%",
                                                      }
                                        }
                                />
                        </span>
                }
                showMenuIconButton={true}
                title={false}
                zDepth={1}
        />
</div>,
      "style": Object {
        "height": 64,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <AppBar
          iconElementLeft={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                  "maxWidth": 50,
                                                }
                              }
                    >
                              <DatePicker
                                        autoOk={false}
                                        container="inline"
                                        disableYearSelection={false}
                                        disabled={false}
                                        firstDayOfWeek={1}
                                        hideCalendarDate={false}
                                        id="calendarNav"
                                        inputStyle={
                                                  Object {
                                                            "color": "white",
                                                          }
                                        }
                                        mode="landscape"
                                        onChange={[Function]}
                                        openToYearSelection={false}
                                        style={Object {}}
                                        value={2017-06-13T11:41:20.000Z}
                              />
                    </div>
          }
          iconElementRight={
                    <span>
                              <FlatButton
                                        disabled={false}
                                        fullWidth={false}
                                        label={
                                                  <FontIcon
                                                            className="fa fa-search"
                                                            color="#ffffff"
                                                            onMouseEnter={[Function]}
                                                            onMouseLeave={[Function]}
                                                  />
                                        }
                                        labelPosition="after"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "height": "100%",
                                                          }
                                        }
                              />
                              <FlatButton
                                        disabled={false}
                                        fullWidth={false}
                                        label={
                                                  <FontIcon
                                                            className="fa fa-plus"
                                                            color="#ffffff"
                                                            onMouseEnter={[Function]}
                                                            onMouseLeave={[Function]}
                                                  />
                                        }
                                        labelPosition="after"
                                        labelStyle={Object {}}
                                        onClick={[Function]}
                                        onKeyPress={[Function]}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onTouchStart={[Function]}
                                        primary={false}
                                        secondary={false}
                                        style={
                                                  Object {
                                                            "height": "100%",
                                                          }
                                        }
                              />
                    </span>
          }
          showMenuIconButton={true}
          title={false}
          zDepth={1}
/>,
        "style": Object {
          "position": "fixed",
          "width": "100%",
        },
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "iconElementLeft": <div
            style={
                        Object {
                                    "height": "100%",
                                    "maxWidth": 50,
                                  }
            }
>
            <DatePicker
                        autoOk={false}
                        container="inline"
                        disableYearSelection={false}
                        disabled={false}
                        firstDayOfWeek={1}
                        hideCalendarDate={false}
                        id="calendarNav"
                        inputStyle={
                                    Object {
                                                "color": "white",
                                              }
                        }
                        mode="landscape"
                        onChange={[Function]}
                        openToYearSelection={false}
                        style={Object {}}
                        value={2017-06-13T11:41:20.000Z}
            />
</div>,
          "iconElementRight": <span>
            <FlatButton
                        disabled={false}
                        fullWidth={false}
                        label={
                                    <FontIcon
                                                className="fa fa-search"
                                                color="#ffffff"
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                    />
                        }
                        labelPosition="after"
                        labelStyle={Object {}}
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        primary={false}
                        secondary={false}
                        style={
                                    Object {
                                                "height": "100%",
                                              }
                        }
            />
            <FlatButton
                        disabled={false}
                        fullWidth={false}
                        label={
                                    <FontIcon
                                                className="fa fa-plus"
                                                color="#ffffff"
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                    />
                        }
                        labelPosition="after"
                        labelStyle={Object {}}
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onTouchStart={[Function]}
                        primary={false}
                        secondary={false}
                        style={
                                    Object {
                                                "height": "100%",
                                              }
                        }
            />
</span>,
          "showMenuIconButton": true,
          "title": false,
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "position": "fixed",
                              "width": "100%",
                            }
          }
>
          <AppBar
                    iconElementLeft={
                              <div
                                        style={
                                                  Object {
                                                            "height": "100%",
                                                            "maxWidth": 50,
                                                          }
                                        }
                              >
                                        <DatePicker
                                                  autoOk={false}
                                                  container="inline"
                                                  disableYearSelection={false}
                                                  disabled={false}
                                                  firstDayOfWeek={1}
                                                  hideCalendarDate={false}
                                                  id="calendarNav"
                                                  inputStyle={
                                                            Object {
                                                                      "color": "white",
                                                                    }
                                                  }
                                                  mode="landscape"
                                                  onChange={[Function]}
                                                  openToYearSelection={false}
                                                  style={Object {}}
                                                  value={2017-06-13T11:41:20.000Z}
                                        />
                              </div>
                    }
                    iconElementRight={
                              <span>
                                        <FlatButton
                                                  disabled={false}
                                                  fullWidth={false}
                                                  label={
                                                            <FontIcon
                                                                      className="fa fa-search"
                                                                      color="#ffffff"
                                                                      onMouseEnter={[Function]}
                                                                      onMouseLeave={[Function]}
                                                            />
                                                  }
                                                  labelPosition="after"
                                                  labelStyle={Object {}}
                                                  onClick={[Function]}
                                                  onKeyPress={[Function]}
                                                  onKeyboardFocus={[Function]}
                                                  onMouseEnter={[Function]}
                                                  onMouseLeave={[Function]}
                                                  onTouchStart={[Function]}
                                                  primary={false}
                                                  secondary={false}
                                                  style={
                                                            Object {
                                                                      "height": "100%",
                                                                    }
                                                  }
                                        />
                                        <FlatButton
                                                  disabled={false}
                                                  fullWidth={false}
                                                  label={
                                                            <FontIcon
                                                                      className="fa fa-plus"
                                                                      color="#ffffff"
                                                                      onMouseEnter={[Function]}
                                                                      onMouseLeave={[Function]}
                                                            />
                                                  }
                                                  labelPosition="after"
                                                  labelStyle={Object {}}
                                                  onClick={[Function]}
                                                  onKeyPress={[Function]}
                                                  onKeyboardFocus={[Function]}
                                                  onMouseEnter={[Function]}
                                                  onMouseLeave={[Function]}
                                                  onTouchStart={[Function]}
                                                  primary={false}
                                                  secondary={false}
                                                  style={
                                                            Object {
                                                                      "height": "100%",
                                                                    }
                                                  }
                                        />
                              </span>
                    }
                    showMenuIconButton={true}
                    title={false}
                    zDepth={1}
          />
</div>,
        "style": Object {
          "height": 64,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <AppBar
            iconElementLeft={
                        <div
                                    style={
                                                Object {
                                                            "height": "100%",
                                                            "maxWidth": 50,
                                                          }
                                    }
                        >
                                    <DatePicker
                                                autoOk={false}
                                                container="inline"
                                                disableYearSelection={false}
                                                disabled={false}
                                                firstDayOfWeek={1}
                                                hideCalendarDate={false}
                                                id="calendarNav"
                                                inputStyle={
                                                            Object {
                                                                        "color": "white",
                                                                      }
                                                }
                                                mode="landscape"
                                                onChange={[Function]}
                                                openToYearSelection={false}
                                                style={Object {}}
                                                value={2017-06-13T11:41:20.000Z}
                                    />
                        </div>
            }
            iconElementRight={
                        <span>
                                    <FlatButton
                                                disabled={false}
                                                fullWidth={false}
                                                label={
                                                            <FontIcon
                                                                        className="fa fa-search"
                                                                        color="#ffffff"
                                                                        onMouseEnter={[Function]}
                                                                        onMouseLeave={[Function]}
                                                            />
                                                }
                                                labelPosition="after"
                                                labelStyle={Object {}}
                                                onClick={[Function]}
                                                onKeyPress={[Function]}
                                                onKeyboardFocus={[Function]}
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                                onTouchStart={[Function]}
                                                primary={false}
                                                secondary={false}
                                                style={
                                                            Object {
                                                                        "height": "100%",
                                                                      }
                                                }
                                    />
                                    <FlatButton
                                                disabled={false}
                                                fullWidth={false}
                                                label={
                                                            <FontIcon
                                                                        className="fa fa-plus"
                                                                        color="#ffffff"
                                                                        onMouseEnter={[Function]}
                                                                        onMouseLeave={[Function]}
                                                            />
                                                }
                                                labelPosition="after"
                                                labelStyle={Object {}}
                                                onClick={[Function]}
                                                onKeyPress={[Function]}
                                                onKeyboardFocus={[Function]}
                                                onMouseEnter={[Function]}
                                                onMouseLeave={[Function]}
                                                onTouchStart={[Function]}
                                                primary={false}
                                                secondary={false}
                                                style={
                                                            Object {
                                                                        "height": "100%",
                                                                      }
                                                }
                                    />
                        </span>
            }
            showMenuIconButton={true}
            title={false}
            zDepth={1}
/>,
          "style": Object {
            "position": "fixed",
            "width": "100%",
          },
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "iconElementLeft": <div
              style={
                            Object {
                                          "height": "100%",
                                          "maxWidth": 50,
                                        }
              }
>
              <DatePicker
                            autoOk={false}
                            container="inline"
                            disableYearSelection={false}
                            disabled={false}
                            firstDayOfWeek={1}
                            hideCalendarDate={false}
                            id="calendarNav"
                            inputStyle={
                                          Object {
                                                        "color": "white",
                                                      }
                            }
                            mode="landscape"
                            onChange={[Function]}
                            openToYearSelection={false}
                            style={Object {}}
                            value={2017-06-13T11:41:20.000Z}
              />
</div>,
            "iconElementRight": <span>
              <FlatButton
                            disabled={false}
                            fullWidth={false}
                            label={
                                          <FontIcon
                                                        className="fa fa-search"
                                                        color="#ffffff"
                                                        onMouseEnter={[Function]}
                                                        onMouseLeave={[Function]}
                                          />
                            }
                            labelPosition="after"
                            labelStyle={Object {}}
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            onKeyboardFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onTouchStart={[Function]}
                            primary={false}
                            secondary={false}
                            style={
                                          Object {
                                                        "height": "100%",
                                                      }
                            }
              />
              <FlatButton
                            disabled={false}
                            fullWidth={false}
                            label={
                                          <FontIcon
                                                        className="fa fa-plus"
                                                        color="#ffffff"
                                                        onMouseEnter={[Function]}
                                                        onMouseLeave={[Function]}
                                          />
                            }
                            labelPosition="after"
                            labelStyle={Object {}}
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            onKeyboardFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onTouchStart={[Function]}
                            primary={false}
                            secondary={false}
                            style={
                                          Object {
                                                        "height": "100%",
                                                      }
                            }
              />
</span>,
            "showMenuIconButton": true,
            "title": false,
            "zDepth": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
